rust/worker/src/index/metadata/types.rs

```diff
@@ -1,190 +1,494 @@
+                    rbm.insert(offset_id);
+                }
+                _ => return Err(Box::new(MetadataIndexError::InvalidKeyType)),
+            },
+            MetadataIndexWriter::F32MetadataIndexWriter(_, uncommitted_rbms) => match key {
+                KeyWrapper::Float32(k) => {
+                    let rbms = uncommitted_rbms.get_mut(prefix).unwrap();
+                    let rbm = rbms.iter_mut().find(|(rbm_k, _)| *rbm_k == k).unwrap();
+                    rbm.1.insert(offset_id);
+                }
+                _ => return Err(Box::new(MetadataIndexError::InvalidKeyType)),
+            },
+        }
         Ok(())
     }
 
     pub async fn write_to_blockfile(&mut self) -> Result<(), Box<dyn ChromaError>> {
-        for (key, mut rbms) in self.uncommitted_rbms.drain() {
-            for (value, rbm) in rbms.drain(..) {
-                self.blockfile_writer.set(key.as_str(), value, &rbm).await?;
+        match self {
+            MetadataIndexWriter::StringMetadataIndexWriter(blockfile_writer, uncommitted_rbms) => {
+                for (prefix, rbms) in uncommitted_rbms.drain() {
+                    for (key, rbm) in rbms.iter() {
+                        blockfile_writer
+                            .set(prefix.as_str(), key.as_str(), rbm)
+                            .await?
+                    }
+                }
+            }
+            MetadataIndexWriter::U32MetadataIndexWriter(blockfile_writer, uncommitted_rbms) => {
+                for (prefix, rbms) in uncommitted_rbms.drain() {
+                    for (key, rbm) in rbms.iter() {
+                        blockfile_writer.set(prefix.as_str(), *key, rbm).await?
+                    }
+                }
+            }
+            MetadataIndexWriter::F32MetadataIndexWriter(blockfile_writer, uncommitted_rbms) => {
+                for (prefix, rbms) in uncommitted_rbms.drain() {
+                    for (key, rbm) in rbms.iter() {
+                        blockfile_writer.set(prefix.as_str(), *key, rbm).await?
+                    }
+                }
+            }
+            MetadataIndexWriter::BoolMetadataIndexWriter(blockfile_writer, uncommitted_rbms) => {
+                for (prefix, rbms) in uncommitted_rbms.drain() {
+                    for (key, rbm) in rbms.iter() {
+                        blockfile_writer.set(prefix.as_str(), *key, rbm).await?
+                    }
+                }
             }
         }
-        self.uncommitted_rbms.clear();
         Ok(())
     }
 
-    pub async fn commit(self) -> Result<MetadataIndexFlusher<K>, Box<dyn ChromaError>> {
-        let f = match self.blockfile_writer.commit::<K, &RoaringBitmap>() {
-            Ok(f) => f,
-            Err(e) => return Err(e),
-        };
-        Ok(MetadataIndexFlusher {
-            phantom: PhantomData,
-            flusher: f,
-        })
+    pub async fn commit(self) -> Result<MetadataIndexFlusher, Box<dyn ChromaError>> {
```

Comment:
why not just forward the generics? instead of using enums?

commit<>? That is the pattern we use elsewhere.